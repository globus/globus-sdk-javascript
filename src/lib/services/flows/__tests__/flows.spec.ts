import serviceTestSuite from "../../../../__utils__/service-test-suite";
import { flows } from "..";

serviceTestSuite("flows", "flows", (fetch) => {
  test("getAll", async () => {
    await flows.getAll();
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(`https://flows.globus.org/flows`, {
      headers: {},
    });
  });

  test("getAll - with headers", async () => {
    await flows.getAll({
      headers: {
        Authorization: "Bearer this-is-an-example-token",
      },
    });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(`https://flows.globus.org/flows`, {
      headers: {
        Authorization: "Bearer this-is-an-example-token",
      },
    });
  });

  test("getAll - with headers and fetch overriders", async () => {
    await flows.getAll(
      {
        headers: {
          Authorization: "Bearer this-is-an-example-token",
        },
      },
      {
        fetch: {
          options: {
            headers: {
              "X-Test-Header": "test",
            },
          },
        },
      }
    );
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(`https://flows.globus.org/flows`, {
      headers: {
        Authorization: "Bearer this-is-an-example-token",
        "X-Test-Header": "test",
      },
    });
  });
});
