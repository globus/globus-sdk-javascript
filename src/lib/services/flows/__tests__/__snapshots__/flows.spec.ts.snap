// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flows.flow getAll - with headers 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "authorization": "Bearer this-is-an-example-token",
    "connection": "close",
    "host": "flows.globus.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "GET",
  "url": "https://flows.globus.org/flows",
}
`;

exports[`flows.flow getAll - with headers and fetch overriders 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "authorization": "Bearer this-is-an-example-token",
    "connection": "close",
    "host": "flows.globus.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
    "x-test-header": "test",
  },
  "method": "GET",
  "url": "https://flows.globus.org/flows",
}
`;

exports[`flows.flow getAll 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "host": "flows.globus.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "GET",
  "url": "https://flows.globus.org/flows",
}
`;

exports[`flows.flow run 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "authorization": "Bearer this-is-an-example-token",
    "connection": "close",
    "content-length": "78",
    "content-type": "application/json",
    "host": "flows.globus.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "json": {
    "body": {
      "echo_string": "Hello, world!",
      "sleep_time": 5,
    },
    "tags": [
      "tag1",
      "tag2",
    ],
  },
  "method": "POST",
  "url": "https://flows.globus.org/flows/flow-id/run",
}
`;
